daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Node.scala

```diff
@@ -81,6 +81,11 @@ object Node {
       chosenValue: Val,
       stakeholders: Set[Party],
       signatories: Set[Party],
+      /** Note that here we have both actors and controllers because we use this
+        * data structure _before_ we validate the transaction. However for every
+        * valid transaction the actors must be the same as the controllers. This
+        * is why we removed the controllers field in transaction version 6.
+        */
       controllers: Set[Party],
       children: ImmArray[Nid])
       extends GenNode[Nid, Cid, Val] {
```

Comment:
And this is more for type-level documentation of a fact of our model; it might make porting nicer, but you might consider it more confusing than helpful, whichever way you think:

```suggestion
       extends GenNode[Nid, Cid, Val] {
         def controllers: actingParties.type = actingParties
```